ext/PathfinderTuringExt.jl

```diff
@@ -2,149 +2,105 @@ module PathfinderTuringExt
+    create_log_density_problem(model::DynamicPPL.Model)
 
-Get `Dict` mapping variable names in model to their ranges in a corresponding parameter vector.
+Create a log density problem from a `model`.
 
-# Examples
+The return value is an object implementing the LogDensityProblems API whose log-density is
+that of the `model` transformed to unconstrained space with the appropriate log-density
+adjustment due to change of variables.
+"""
+function create_log_density_problem(model::DynamicPPL.Model)
+    # create an unconstrained VarInfo
+    varinfo = DynamicPPL.link(DynamicPPL.VarInfo(model), model)
+    # DefaultContext ensures that the log-density adjustment is computed
+    prob = DynamicPPL.LogDensityFunction(varinfo, model, DynamicPPL.DefaultContext())
+    return prob
+end
 
-```julia
-julia> @model function demo()
-           s ~ Dirac(1)
-           x = Matrix{Float64}(undef, 2, 4)
-           x[1, 1] ~ Dirac(2)
-           x[2, 1] ~ Dirac(3)
-           x[3] ~ Dirac(4)
-           y ~ Dirac(5)
-           x[4] ~ Dirac(6)
-           x[:, 3] ~ arraydist([Dirac(7), Dirac(8)])
-           x[[2, 1], 4] ~ arraydist([Dirac(9), Dirac(10)])
-           return s, x, y
-       end
-demo (generic function with 2 methods)
+"""
+    draws_to_chains(model::DynamicPPL.Model, draws) -> MCMCChains.Chains
 
-julia> demo()()
-(1, Any[2.0 4.0 7 10; 3.0 6.0 8 9], 5)
+Convert a `(nparams, ndraws)` matrix of unconstrained `draws` to an `MCMCChains.Chains`
+object with corresponding constrained draws and names according to `model`.
+"""
+function draws_to_chains(model::DynamicPPL.Model, draws::AbstractMatrix)
+    varinfo = DynamicPPL.link(DynamicPPL.VarInfo(model), model)
+    draw_con_varinfos = map(eachcol(draws)) do draw
+        # this re-evaluates the model but allows supporting dynamic bijectors
+        # https://github.com/TuringLang/Turing.jl/issues/2195
+        return Turing.Inference.getparams(model, DynamicPPL.unflatten(varinfo, draw))
+    end
+    param_con_names = map(first, first(draw_con_varinfos))
+    draws_con = reduce(
+        vcat, Iterators.map(transpose ∘ Base.Fix1(map, last), draw_con_varinfos)
+    )
+    chns = MCMCChains.Chains(draws_con, param_con_names)
+    return chns
+end
 
-julia> varnames_to_ranges(demo())
-Dict{AbstractPPL.VarName, UnitRange{Int64}} with 8 entries:
-  s           => 1:1
-  x[4]        => 5:5
-  x[:,3]      => 6:7
-  x[1,1]      => 2:2
-  x[2,1]      => 3:3
-  x[[2, 1],4] => 8:9
-  x[3]        => 4:4
-  y           => 10:10
-```
 """
-function varnames_to_ranges end
+    transform_to_constrained(
+        p::AbstractArray, vi::DynamicPPL.VarInfo, model::DynamicPPL.Model
+    )
 
-varnames_to_ranges(model::DynamicPPL.Model) = varnames_to_ranges(DynamicPPL.VarInfo(model))
-function varnames_to_ranges(varinfo::DynamicPPL.UntypedVarInfo)
-    return varnames_to_ranges(varinfo.metadata)
+Transform a vector of parameters `p` from unconstrained to constrained space.
+"""
+function transform_to_constrained(
```

Comment:
Is this function still used or could it be removed?

Reply:
Sure, can absolutely be removed.